### PR TITLE
Created StatusBox component to hold messages and errors

### LINKED ISSUE #416: Refactor React message code into single class

### PR DESCRIPTION
**Pull request by noisecapella**
Closes #416 

I also added `forceUpdate` at the top of the PR to fix a minor testing bug where async state updates would not be caught.

### CODE REVIEW

**Comment by pdpinch** [comment]
George - is there an issue that goes with this?

Also, how does this interact with django messages?

**Comment by noisecapella** [comment]
I just attached an issue. It doesn't interact with django messages at all, it's just a React component to provide an elegant way to show the same bootstrap error styling everywhere

File: ui/static/ui/js/utils.jsx [hunk 1]
```diff
@@ -23,6 +23,34 @@ define("utils", ["jquery", "lodash"], function ($, _) {
...
+        return <div className="alert alert-danger alert-dismissible">
+          {this.props.message.error}
+        </div>;
+      }
+
+      if (this.props.message !== undefined) {
+        return <div className="alert alert-success alert-dismissible">
+          {this.props.message}
+          </div>;
+      }
```
Comment (giocalitri) [inline ui/static/ui/js/utils.jsx:48]: Is there any part in the code where you show success messages?

Comment (noisecapella) [inline ui/static/ui/js/utils.jsx:48]: The previous few lines show `this.props.message` in a `alert-success` div

Comment (giocalitri) [inline ui/static/ui/js/utils.jsx:48]: Yes, I understood, but is there anywhere in the code a situation where we actually show an `alert-success` div?
Just asking to test it.

Comment (noisecapella) [inline ui/static/ui/js/utils.jsx:48]: I see. If you open a learning resource and save the form it should give you the success message there

**Comment by giocalitri** [comment]
Looks good to me: functionality is fine. Merging

**Pull request merged**